src/Parsers/ASTFunction.h

```diff
@@ -22,6 +22,8 @@ class ASTFunction : public ASTWithAlias
 
     bool is_window_function = false;
 
+    bool compute_after_window_functions = false;
```

Comment:
AST Looks like not the best place for this flag, but we already have `is_window_function` here, so it seems now there's no easy way to get rid of it. And we need actually to determine which expressions are related to the window in `WindowExpressionsCollectorChildInfo` that also operates with AST, though

Reply:
In my opinion, it's completely okay to cache some useful data in AST. Instead of that we usually recompute the same values several times.

The only part that should be considered is that we don't want to make AST nodes too big. In this case, it's not a problem, because I just added a simple boolean flag (in the opposite situation we could store a pointer to a struct with cached values).